app/schema/widget.py

```diff
@@ -0,0 +1,9 @@
+class Widget(object):
+    def __init__(self, name):
+        self.name = name
+
+    def render(self, schema, state):
+        pass
```

Comment:
Can we tighten up this interface contract so that we return an NotImplemented exception if (and only if) we expect all widgets to implement the `render` and `get_user_input` methods? For reference on the exception: https://dwieeb.com/2015/06/02/usages-of-notimplemented-and-notimplementederror-in-python/